src/test/java/com/shapesecurity/salvation/LocationTest.java

```diff
@@ -306,6 +306,54 @@
         assertEquals(27, notices.get(1).endLocation.offset);
     }
 
+    @Test public void testWarningLocationRequireSriFor() {
+        ArrayList<Notice> notices = new ArrayList<>();
+
+        ParserWithLocation.parse("require-sri-for", "https://origin", notices);
```

Comment:
If we don't change the spec grammar, this remains a valid policy, so I think warning is appropriate here.